### DISCUSSION TITLE
v0.3.0

### DISCUSSION THREAD

**Discussion by pajowu** [bugbakery/audapolis#449] [Announcements]
After a long time of development, we have big news today: Version 0.3.0 is now available, and we changed our organization-name 🎉.

> **Warning**: If you save your file using this version, you will **no longer be able to open it in older versions of audapolis**.

You can download this version for the following platforms:
* Linux in the following formats
    * [.AppImage](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-linux-x86_64-0.3.0_4.AppImage) 
    * [.deb](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-linux-amd64-0.3.0_4.deb)
    * [.rpm](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-linux-x86_64-0.3.0_4.rpm)
    * [.pacman](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-linux-x64-0.3.0_4.pacman)
* Mac OS
    > **Note:** If your mac shows a popup that this app could not be verified with apple and refuses to start the app, try launching it by right clicking on the app icon and then selecting "Open".
    * [For Intel Chips (works on all modern Macs)](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-mac-x64-0.3.0_4.dmg)
    * [For Apple Silicon (only works on Macs with an M1 CPU)](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-mac-arm64-0.3.0_4.dmg)
* [Windows](https://github.com/bugbakery/audapolis/releases/download/v0.3.0/audapolis-win-x64-0.3.0_4.exe) 

## New Name, New Project, New Me?

We, the team behind audapolis changed our name. We are now named "The Vegan Bug Bakery", no longer "the audapolis team". This has a simple reason: We grew and started another project – [transcribee 🐝](https://github.com/bugbakery/transcribee).

Transcribee is based on a lot of feedback we got from audapolis. Mainly the need of many people to have an easy-to-use transcription software. With a new funding from the [prototype fund](https://prototypefund.de) – who also funded the audapolis development – we now try to meet this need.

If you are interested in transcribee, you can take a look [at the transcribee github repo](https://github.com/bugbakery/transcribee). Transcribee is a collaborative web-platform that is intended to be hosted by organizations (e.g. universities) and solely focuses on transcription and transcript correction. With this, it can provide much better out of the box transcripts and experience editing the transcripts at the cost of not having the editing features of audapolis.

If you want to learn more about our work, you can follow our [twitter](https://twitter.com/bugbakery), [mastodon](https://chaos.social/@bugbakery) or [github](https://github.com/bugbakery) profiles.

## What's Changed

But just because we have a new project, doesn't mean we don't work on audapolis anymore. Instead, we added some bigger (and a whole lot of smaller) changes to this new version.

In total, we added 118 commits with 12,343 additions and 6,794 deletions.

Special thanks to @lucaswerkmeister and @phlmn, who did their first contributions to audapolis in this release.

### New Fileformat

The biggest change is probably invisible to most of you: We changed the internal file format that audapolis uses to store the data. We hope that this new format allows us to improve the performance of audapolis in many places and reduce some bugs. For the first time, we also wrote a longer document describing this file format, so other apps can use it too.

* ✨ Fileformat v3 by @pajowu in https://github.com/bugbakery/audapolis/pull/354

### Document Filter

Have you ever wanted to create a supercut of every time someone says "apple" in a meeting?
Or you need all paragraphs that contain a certain phrase? Now you can do this is no time with the new "Document Filter" action. This action can be found in the menu under "Edit" or by pressing <kbd>Shift</kbd><kbd>Ctrl</kbd><kbd>F</kbd> / <kbd>Shift</kbd><kbd>⌘</kbd><kbd>F</kbd>.

This feature was inspired by the incredible [videogrep](https://github.com/antiboredom/videogrep) project.

* ✨ Add document filter action by @pajowu in https://github.com/bugbakery/audapolis/pull/367

### Export Debug Logs

While we try to keep audapolis as bug-free as possible, sometimes we miss something. This is why we introduce **better logging**. Users can now export debug logs from the "Help" menu, which help us better understand some bugs.

* Log to file & add option to export debug logs by @pajowu in https://github.com/bugbakery/audapolis/pull/372

### UX Improvements

We fixed a number of small annoyances we and other users found when using audapolis:

* You can now change the font size using <kbd>Ctrl</kbd><kbd>+</kbd> and <kbd>Ctrl</kbd><kbd>-</kbd> / <kbd>⌘</kbd><kbd>+</kbd> and <kbd>⌘</kbd><kbd>-</kbd> (@pajowu in https://github.com/bugbakery/audapolis/pull/400)
* We replaced the hover-menu in the editor with a proper context menu (@anuejn in https://github.com/bugbakery/audapolis/pull/341)
* 📽️ use normal export dialog for selection export by @anuejn in https://github.com/bugbakery/audapolis/pull/315
* ✨ Plain text export by @pajowu in https://github.com/bugbakery/audapolis/pull/321
* Add webm to importing extensions by @lucaswerkmeister in https://github.com/bugbakery/audapolis/pull/328
* ✨ Allow cancellation of model downloads by @pajowu in https://github.com/bugbakery/audapolis/pull/338
* 🩹 Add icon in more formats to show on linux by @pajowu in https://github.com/bugbakery/audapolis/pull/399
* 🐬 add app-icon padding on macOS and linux by @anuejn in https://github.com/bugbakery/audapolis/pull/434

### Transcript Correction

You can now edit the text of the transcript more easily by pressing <kbd>i</kbd> / <kbd>o</kbd> to change the word left/right of the cursor. If you select parts of the text and press either key, you can edit a longer chunk of text at once.

* better transcript correction by @anuejn in https://github.com/bugbakery/audapolis/pull/332

### Documentation

We didn't just work on the code, but also on the surrounding documentation.

We added contribution guidelines and a code of conduct. The [contribution guidelines](https://github.com/bugbakery/audapolis/blob/main/CONTRIBUTING.md) now explain how you can contribute to audapolis, be it through reporting bugs or unclear documentation, suggesting new features or writing code. The [code of conduct](https://github.com/bugbakery/audapolis/blob/main/CODE_OF_CONDUCT.md) explains what kind of community we want to create with audapolis and which behavior we find acceptable – and which not.

We also added information on how you can support us financially, so we can spend more time on audapolis.

* 📝 Add contribution guidelines by @pajowu in https://github.com/bugbakery/audapolis/pull/352
* 📝 Update download links in readme by @pajowu in https://github.com/bugbakery/audapolis/pull/313
* 🤑 Create FUNDING.yml by @pajowu in https://github.com/bugbakery/audapolis/pull/342
* Create CODE_OF_CONDUCT.md by @pajowu in https://github.com/bugbakery/audapolis/pull/348

### 🚧 Punctuation Reconstruction

We added an optional setting to enable punctuation reconstruction, i.e. trying to guess where periods, commas etc. were in the spoken text. This feature is still very experimental. If you find any bugs, please report them to us.

* 🚧 Punctuation reconstruction by @pajowu in https://github.com/bugbakery/audapolis/pull/330
* Fix aesara crash on windows by @pajowu in https://github.com/bugbakery/audapolis/pull/427

### Auto Updates

Starting with v0.2.2-pre2, audapolis checks if a new version was released and then asks you if this new version should be installed.

* ✨ Auto-Updates by @pajowu in https://github.com/bugbakery/audapolis/pull/319

### Miscellaneous

We fixed some bugs

* 🐛 Fix undo working across documents by @pajowu in https://github.com/bugbakery/audapolis/pull/351
* 🐛 Fix bug that prevented user from disabling punctuation by @pajowu in https://github.com/bugbakery/audapolis/pull/363
* 🐛 Fix save dialog not including a default extension on some platforms by @pajowu in https://github.com/bugbakery/audapolis/pull/353
* 🚸 Improve "stop model download" ui by @pajowu in https://github.com/bugbakery/audapolis/pull/369
* 🚑️ Use poetry export -o instead of shell redirect by @pajowu in https://github.com/bugbakery/audapolis/pull/384
* 🐛 Fix crashes when pressing Esc in comboboxes by @pajowu in https://github.com/bugbakery/audapolis/pull/386
* 🐛 Run log arguments to string conversion before ipc by @pajowu in https://github.com/bugbakery/audapolis/pull/387
* Do not split render items on speaker change by @pajowu in https://github.com/bugbakery/audapolis/pull/393
* eslint: Fix typo 'detetect' by @phlmn in https://github.com/bugbakery/audapolis/pull/423
* 🐛 Fix crash if "Esc" after interacting with export dialog by @pajowu in https://github.com/bugbakery/audapolis/pull/318
* 🐛 Do not allow selection to be outside document by @pajowu in https://github.com/bugbakery/audapolis/pull/325
* 🐛 Correctly calculate time after last element by @pajowu in https://github.com/bugbakery/audapolis/pull/339
* 🐛 Fix export options now showing by @pajowu in https://github.com/bugbakery/audapolis/pull/349

---

We updated our dependencies

* Bump nanoid from 3.1.25 to 3.3.1 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/331
* Bump ejs from 3.1.6 to 3.1.8 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/362
* Bump plist from 3.0.4 to 3.0.5 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/361
* Bump async from 2.6.3 to 2.6.4 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/364
* Bump ansi-regex from 3.0.0 to 3.0.1 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/360
* Bump electron from 13.5.2 to 13.6.6 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/356
* Bump shell-quote from 1.7.2 to 1.7.3 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/383
* Bump vite from 2.5.7 to 2.9.13 in /app by @dependabot in https://github.com/bugbakery/audapolis/pull/382
* ⬆️ Upgrade evergreen version by @pajowu in https://github.com/bugbakery/audapolis/pull/368
* ⬆️ Bump webvtt-writer by @pajowu in https://github.com/bugbakery/audapolis/pull/391
* ⬆️ Upgrade python dependencies, enables m1 builds by @pajowu in https://github.com/bugbakery/audapolis/pull/388
* ⬆️ Bump punctuator2 version by @pajowu in https://github.com/bugbakery/audapolis/pull/398

---

We updated some internal tools

* 🩹 Update transcribe.py by @pajowu in https://github.com/bugbakery/audapolis/pull/408
* 🐛 Correctly handle no punctuation in transcribe script by @pajowu in https://github.com/bugbakery/audapolis/pull/412

---

We fought with our CI

* 🔧 update ci setup dependencies by @anuejn in https://github.com/bugbakery/audapolis/pull/424
* 💚 Checkout with full history for build by @pajowu in https://github.com/bugbakery/audapolis/pull/345
* 💚 Version with . instead of + by @pajowu in https://github.com/bugbakery/audapolis/pull/346
* 💚 Use github safe version string for artifact name by @pajowu in https://github.com/bugbakery/audapolis/pull/347
* 📦 Use + instead of - for versions by @pajowu in https://github.com/bugbakery/audapolis/pull/343
* 👷‍♀️ modernize ci by @anuejn in https://github.com/bugbakery/audapolis/pull/435
* 🙈 fixup: modernize ci by @anuejn in https://github.com/bugbakery/audapolis/pull/437
* 💚 Rebuild if build.yml changed by @pajowu in https://github.com/bugbakery/audapolis/pull/438
* 💚 Remove set-output usage by @pajowu in https://github.com/bugbakery/audapolis/pull/440
* 💚 Fix macos-latest build hanging on signing cert by @pajowu in https://github.com/bugbakery/audapolis/pull/441
* 🏗️ Replace realpath with readlink -f by @pajowu in https://github.com/bugbakery/audapolis/pull/442

---

We changed the name of the github orga: 

* ♻️ Rename orga to bugbakery by @pajowu in https://github.com/bugbakery/audapolis/pull/443

---

And last but not least, we released this new version

* Bump version to 0.3.0 by @pajowu in https://github.com/bugbakery/audapolis/pull/439

**Full Changelog**: https://github.com/bugbakery/audapolis/compare/v0.2.1-1...v0.3.0+3

<hr /><em>This discussion was created from the release <a href='https://github.com/bugbakery/audapolis/releases/tag/v0.3.0%2B4'>v0.3.0</a>.</em>

**Comment by jeonsaux24** [comment]
The app doesn't wanna launch on my Windows 10 OS. I wanted to use audapolis for my podcast creation. But it doesn't wanna launch.
